lib/auth/init.go

```diff
@@ -963,6 +963,25 @@ type Identity struct {
 	XCert *x509.Certificate
 	// ClusterName is a name of host's cluster
 	ClusterName string
+	// SystemRoles is a list of additional system roles.
+	SystemRoles []string
```

Comment:
Does it make sense for this to live inside `IdentityID`

Reply:
While the data stored in `IdentityID` is _similar_, the `IdentityID` type is used like the "key" in the key-value credential storage model, and is pre-constructed by agents prior to knowing any broader cluster-wide information.  In contrast, the contents of `SystemRoles` is part of the "value" in the key-value credential storage model (i.e. the same key could load certs with various `SystemRoles` lists), and its exact contents aren't known in advance.  It's the auth-side join token resources that actually determines the contents of this list.  Your instance cert gets whatever system roles your join token grants, meaning that an agent can't infer the list by observing its own local configuration.